source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -389,6 +365,66 @@ changed through an ACL - since it would have a lower priority.
 Access to logical switch ports with router or localnet type **cannot**
 be controlled through ACLs.
 
+## The remote_group_id parameter mapping
+
+The **remote_group_id** security group rule parameter provides the user a
+way to allow ingress/egress traffic for all the VMs attached to ports having
+that security group. It enables the user to leverage semantics like a rule
+allowing all ssh ingress traffic for members of the *ops* security group.
+
+Furthermore, the default Openstack behavior is to create rules allowing ingress
+for all members of the security group at group creation time - including the
+default group. This implies that this parameter is key to fully mimicking how
+openstack behaves.
+
+### How to encode the remote_group_id parameter
+
+The OVN **address set** tables will be used to encode the remote group id
+information.
+
+The OVN address sets are - as per [OVN documentation](http://www.openvswitch.org/support/dist-docs/ovn-nb.5.html) - a 'a named set of addresses'.
+
+Each address set can only feature **one** type of addresses - eth, ipv4, ipv6 -
+and will be used in the ACL match column. Examples of address sets being used in the match conditions can be seen below.
+
+```
+IPv4 match using address sets
+match="ip4 && ip4.src == $set1"
+
+IPv6 match using address sets
+match="ip6 && ip6.src == $set2"
+```
+
+The address sets referenced above would look like:
+```
+ovn-nbctl create Address_Set name=set1 addresses='10.0.0.1 10.0.0.2 10.0.0.3'
+ovn-nbctl create Address_Set name=set2 addresses='2001:db8:0:0:0:0:2:1 2001:db8:0:0:0:0:3:1 2001:db8:0:0:0:0:14:1'
+```
+
+A reference to the associated port group will be stored on the address set's
+external ids.
+
```

Comment:
Does this mean you plan to map the remote_group_id only on L3 addresses?
What would happen if there are no OpenStack subnets?
Would be a possible implementation to define the remote_group_id on L2/eth addresses?

Reply:
OK, you convinced me, that security groups without subnets are quite useless, so L3 appropriate here.

Reply:
That might indeed by a better solution; it is supported (address sets allow eth addresses), and simplifies the scenarios where IP addresses are assigned through DHCP.

One thing that can probably render this possibility useless is that the **remote_group_id** is meant to be used across different networks, and our current provider implementation does not assure that mac addresses are unique (when statically assigned by the user):
```shell
 ✘ mduarted@localhost ~/github/sec_group_demo master ●  vagrant ssh -c 'sudo ovn-nbctl show '
switch cad16023-91ae-4c90-8e8d-1bbf54b58686 (ovirt-demo_net-3724ecff-4cfa-40f3-bec5-6f5aeceaa263)
    port be3f0ce9-f6da-428b-822b-1eb839342e5c
        addresses: ["00:00:aa:00:dd:01 192.168.10.20"]
    port dd7ea492-a7cd-42ea-988c-c80ef33b20e7
        addresses: ["00:00:aa:00:dd:01 192.168.10.10"]
```
The user can assign the same mac to different ports in same networks - which would render L2 networks unusable, but, if he assigns same macs to different ports, in **different** networks, the problem would be located at security group level - we wouldn't be able to discern between both of them.